src/sys/wait.rs

```diff
@@ -270,24 +313,92 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
+/// Designates whether the supplied `Pid` value is a process ID, process group ID,
+/// specifies any child of the current process's group ID, or any child of the current process.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum PidGroup {
+    /// Signifies that the `Pid` is supposed to represent a process ID.
+    ProcessID(Pid),
+    /// Signifies that the `Pid` is supposed to represent a process group ID.
+    ProcessGroupID(Pid),
+    /// Signifies that any child process that belongs to the process group of the current process
+    /// should be selected.
+    AnyGroupChild,
+    /// Signifies that any child process that belongs to the current process should be selected.
+    AnyChild,
+}
+
+impl From<i32> for PidGroup {
```

Comment:
This should be removed as it's unused and we shouldn't support this for our users.